src/Listener/GraphQL/GraphQLMiddlewareContext.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+
+namespace SilverStripe\Snapshots\Listener\GraphQL;
+
+
+use GraphQL\Language\AST\NodeKind;
+use GraphQL\Language\Parser;
+use GraphQL\Language\Source;
+use GraphQL\Type\Schema;
+use SilverStripe\GraphQL\Manager;
+use SilverStripe\Snapshots\Listener\EventContext;
+
+class GraphQLMiddlewareContext extends EventContext
+{
+    /**
+     * @var string
+     */
+    private $query;
+
+    /**
+     * @var Schema|null
+     */
+    private $schema;
+
+    /**
+     * @var array
+     */
+    private $graphqlContext = [];
+
+    /**
+     * @var array
+     */
+    private $params = [];
+
+    /**
+     * GraphQLMiddlewareContext constructor.
+     * @param string $query
+     * @param Schema|null $schema
+     * @param array $graphqlContext
+     * @param array|null $params
+     */
+    public function __construct(string $query, ?Schema $schema = null, array $graphqlContext = [], ?array $params = [])
+    {
+        $this->query = $query;
+        $this->schema = $schema;
+        $this->graphqlContext = $graphqlContext;
+        $this->params = $params;
+    }
+
+    /**
+     * @return string
+     */
+    public function getQuery(): string
+    {
+        return $this->query;
+    }
+
+    /**
+     * @return Schema|null
+     */
+    public function getSchema(): ?Schema
+    {
+        return $this->schema;
+    }
+
+    /**
+     * @return array
+     */
+    public function getGraphqlContext(): array
+    {
+        return $this->graphqlContext;
+    }
+
+    /**
+     * @return array
+     */
+    public function getParams(): array
+    {
+        return $this->params;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAction(): string
+    {
+        $document = Parser::parse(new Source($this->getQuery() ?: 'GraphQL'));
```

Comment:
You've changed the functionality of this function (`getAction`). For example for this input:

```
mutation updateLayoutBlockGroup($block: AppLayoutBlockUpdateInputType!) {
  updateAppLayoutBlock(Input: $block) {
    ID
    TargetingGroup
    TargetingEnabled
    __typename
  }
}
```

Output before:

```
updateLayoutBlockGroup
```

Output after:

```
updateAppLayoutBlock
```

Reply:
Alright, the name of the operation is not required, so we'll have to prefer that, and fall back on the name of the mutation as defined in the schema (what I have now).

Reply:
Have fixed this ^